### PR TITLE
Put Helix manager out of resource manager

### PR DESCRIPTION
**Pull request by jackjlli**
This PR puts Helix manager out of resource manager.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -168,8 +168,8 @@ public PinotHelixResourceManager(@Nonnull ControllerConf controllerConf) {
   /**
    * Create Helix cluster if needed, and then start a Pinot controller instance.
    */
-  public synchronized void start() {
-    _helixZkManager = registerAndConnectAsHelixParticipant();
+  public synchronized void start(HelixManager helixZkManager) {
+    _helixZkManager = helixZkManager;
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Use helixManager.getInstanceName() as the _instanceId

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java [hunk 2]
```diff
@@ -426,6 +432,29 @@ private void initPinotCrypterFactory() {
     }
   }
 
+  /**
+   * Register and connect to Helix cluster as PARTICIPANT role.
+   */
+  private HelixManager registerAndConnectAsHelixParticipant() {
+    String controllerParticipantInstanceId = _helixResourceManager.getInstanceId();
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: Don't fetch instanceId from resource manager, explicitly create one instanceId for controller, one for participant.

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/PerfBenchmarkDriver.java [hunk 3]
```diff
@@ -265,11 +271,32 @@ private void startHelixResourceManager()
...
+      Tenant brokerTenant =
+          new TenantBuilder(_brokerTenantName).setRole(TenantRole.BROKER).setTotalInstances(1).build();
       _helixResourceManager.createBrokerTenant(brokerTenant);
     }
   }
 
+  /**
+   * Register and connect to Helix cluster as Spectator role.
+   */
+  private HelixManager registerAndConnectAsHelixSpectator(ControllerConf controllerConf) {
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/PerfBenchmarkDriver.java]: Pass in instanceId as the parameter

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java (hunk 2 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: Done

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java (hunk 1 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/PerfBenchmarkDriver.java (hunk 3 above)
Comment (jackjlli) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/PerfBenchmarkDriver.java]: Done

**Pull request merged**